### PR TITLE
error diagnostics for unbound identifiers

### PR DESCRIPTION
**Pull request by symphorien**
### Summary & Motivation

find at least as much stuff as nix-instantiate --parse-only

currently not very useful because rnix-lsp cannot parse let .. in nor lambda constructs, and they are present in nearly every file.

image

### CODE REVIEW

**Review by Ma27 — APPROVED** [review]
SGTM. Going to test this for a while locally.

**Comment by Ma27** [comment]
Also would be nice to also get a review from @aaronjanse here :)

**Comment by symphorien** [comment]
I fixed the merge conflict. In another branch ( https://github.com/symphorien/rnix-lsp/tree/static ) I also have implemented support for parsing a large part of nix: let..in, with, string, lists, etc Should I push it here or wait for this PR to be merged ?

**Pull request merged**
